src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,333 @@
+   * Indicates whether an IP address belongs to a subnet. Returns null if any of
+   * the parameters do not represent valid network identifiers.
+   * @param {string} netmask the netmask of the subnet.
+   * @param {string} netIpAddress any IP address on the subnet.
+   * @param {string} testIpAddress the IP address which should be tested for
+   *          whether it belongs to the subnet.
+   * @return {boolean}
+   */
+  public static isInSubnet(netmask: Netmask, netIpAddress: IPv4Address, testIpAddress: IPv4Address): boolean | null {
+    const netmaskInt = this.getIPv4AsInteger(netmask);
+    const netIpInt = this.getIPv4AsInteger(netIpAddress);
+    const testIpInt = this.getIPv4AsInteger(testIpAddress);
+    return ((netmaskInt & netIpInt) === (netmaskInt & testIpInt));
+  }
+
+  /**
+   * Indicates whether an IP address is in a range of IP addresses. Returns null
+   * if any of the parameters do not represent valid network identifiers.
+   * @param {string} ipAddress the IP address which should be tested for whether
+   *          it is in the range.
+   * @param {IPv4AddressRange} range an object of the form { start: "x.x.x.x", end:
+   *          "x.x.x.x" }, representing the range of IP addresses.
+   * @return {boolean}
+   */
+  public static isInIPv4AddressRange(ipAddress: IPv4Address, range: IPv4AddressRange): boolean {
+    const testComponents = ipAddress.address.split('.');
+    const lowComponents = range.start.split('.');
+    const highComponents = range.end.split('.');
+    for (let i = 0; i < 4; i++) {
+      if (!this.inIPv4AddressRange(lowComponents[i], highComponents[i], testComponents[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Takes a set of IP addresses and a set of IP ranges, and indicates whether
+   * any of the IP addresses belong to any of the ranges.
+   * @param {string[]} ips: array of IP strings
+   * @param {IPv4AddressRange[]} ranges: array of range objects
+   * @return {boolean} boolean value
+   */
+  public static isAnyIPv4AddressInAnyIPv4AddressRange(ips: IPv4Address[], ranges: Array<IPv4AddressRange>): boolean {
+    for (let i = 0; i < ranges.length; i++) {
+      if (this.isAnyIPv4AddressInIPv4AddressRange(ips, ranges[i])) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Indicates whether any IP in an array of IPs is within an IP range.
+   * @param {string[]} ips IP address strings.
+   * @param {IPv4AddressRange} range IP range object.
+   * @return {boolean} boolean value
+   */
+  public static isAnyIPv4AddressInIPv4AddressRange(ips: IPv4Address[], range: IPv4AddressRange): boolean {
+    return ips.filter((ip) => {
+      return this.isInIPv4AddressRange(ip, range);
+    }).length > 0;
+  }
+
+  /**
+   * Converts integer to ip address representation.
+   * @param {number} ipInt: Integer to be converted.
+   * @return {string} IP Address string.
+   */
+  public static getIntegerAsIPv4Address(ipInt: number): string {
```

Comment:
can this be private?

Reply:
I don't think so because I'm testing it.